File: Two_Pointers/209.Minimum-Size-Subarray-Sum/209.Minimum-Size-Subarray-Sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int start = 0;
        int end = 0;
        int ret = Integer.MAX_VALUE;
        int windowSum = 0;
        for (; end < nums.length; end++) {
            windowSum += nums[end];
            while (windowSum >= target) {
                ret = Math.min(ret, end - start + 1);
                windowSum -= nums[start++];
            }
        }
        return ret == Integer.MAX_VALUE ? 0 : ret;
    }
}

File: Two_Pointers/209.Minimum-Size-Subarray-Sum/209.Minimum-Size-Subarray-Sum.ts
function minSubArrayLen(target: number, nums: number[]): number {
  let start = 0;
  let end = 0;
  let windowSum = 0;
  let ret = Number.MAX_SAFE_INTEGER;
  for (; end < nums.length; end++) {
    windowSum += nums[end];
    while (windowSum >= target) {
      ret = Math.min(end - start + 1, ret);
      windowSum -= nums[start];
      start += 1;
    }
  }
  return ret === Number.MAX_SAFE_INTEGER ? 0 : ret;
}
